Assets/Scripts/Tile/Connections/AdvancedAdjacencyConnector.cs

```diff
@@ -0,0 +1,216 @@
+        public Mesh xNone;
+        // A mesh where all edges are connected, southeast is a corner
+        public Mesh xSingle;
+        // A mesh where all edges connected, southeast and northeast corners
+        public Mesh xSide;
+        // A mesh where all edges connected, southeast and northwest corners
+        public Mesh xOpposite;
+        // A mesh where all edges connected, all but northwest are corners
+        public Mesh xTriple;
+        // A mesh where all edges connected, all corners
+        public Mesh xQuad;
+
+        public void OnEnable()
+        {
+            SetMeshAndDirection();
+        }
+
+        /**
+         * When a single adjacent turf is updated
+         */
+        public void UpdateSingle(Direction direction, ConstructibleTile tile)
+        {
+            UpdateSingleConnection(direction, tile);
+            SetMeshAndDirection();
+        }
+
+        /**
+         * When all (or a significant number) of adjacent turfs update.
+         * Turfs are ordered by direction, i.e. North, NorthEast, East ... NorthWest
+         */
+        public void UpdateAll(ConstructibleTile[] tiles)
+        {
+            for (int i = 0; i < tiles.Length; i++) {
+                UpdateSingleConnection((Direction)i, tiles[i]);
+            }
+            SetMeshAndDirection();
+        }
+
+        public void Awake()
+        {
+            filter = GetComponent<MeshFilter>();
+        }
+
+        /**
+         * Adjusts the connections value based on the given new tile
+         */
+        private void UpdateSingleConnection(Direction direction, ConstructibleTile tile)
+        {
+            bool isConnected = (tile.turf && (tile.turf.genericType == type || type == null)) || (tile.fixture && (tile.fixture.genericType == type || type == null));
+
+            // Set the direction bit to isConnected (1 or 0)
+            connections &= (byte)~(1 << (int)direction);
+            connections |= (byte)((isConnected ? 1 : 0) << (int)direction);
+        }
+
+        /**
+         * Sorry about the length, but each part is pretty well seperated, so it shouldn't be too difficult to
+         * follow hopefully (-nonanon).
+         * 
+         * This code takes the information about what tiles are adjacent (each of the 8 directions either has an adjacent connectable or not),
+         * and converts this into the mesh to use, along with a direction to rotate the mesh in.
+         */
+        private void SetMeshAndDirection()
+        {
+            // Count number of connections along cardinal, to determine which 'outer' mesh we use of O, C, I/L, T, X
+            int north = Adjacent(Direction.North);
+            int east = Adjacent(Direction.East);
+            int south = Adjacent(Direction.South);
+            int west = Adjacent(Direction.West);
+
+            int numConnections = north + east + south + west;
+
+            float rotation = 0.0f;
+            Mesh mesh;
+            if(numConnections == 0)
```

Comment:
I tried switch, but it starts to get into indentation hell.
Imagine all the ifs that are there, but then in switch-cases you indent twice rather than just once.

I could offload into different functions, but i don't really think it helps the underlying issue of understanding bit-twiddly code

Reply:
Yeah but it would be cleaner. It is already in nesting hell. Eh. Do what you can. You are correct, people will probably avoid this area due to all the bit operations anyway.